contrib/mesos/pkg/election/etcd_master_test.go

```diff
@@ -58,41 +56,71 @@ func TestEtcdMasterNoOther(t *testing.T) {
 	w.Stop()
 }
 
-func TestEtcdMasterNoOtherThenConflict(t *testing.T) {
-	path := "foo"
-	e := tools.NewFakeEtcdClient(t)
-	e.TestIndex = true
-	// Ok, so we set up a chain of responses from etcd:
-	//   1) Nothing there
-	//   2) conflict (someone else wrote)
-	//   3) new value (the data they wrote)
-	empty := tools.EtcdResponseWithError{
-		R: &etcd.Response{
-			Node: nil,
-		},
-		E: &etcd.EtcdError{
-			ErrorCode: tools.EtcdErrorCodeNotFound,
-		},
-	}
-	empty.N = &tools.EtcdResponseWithError{
-		R: &etcd.Response{},
-		E: &etcd.EtcdError{
-			ErrorCode: tools.EtcdErrorCodeNodeExist,
-		},
+// MockClient is implementing the test scenario for TestEtcdMasterNoOtherThenConflict:
+// 1. After first Get() is setting "baz" for the key "foo".
+// 2. Then is expecting Create() call.
+// 3. Then is expecting another Get() call.
+type MockClient struct {
+	client tools.EtcdClient
+	t      *testing.T
+	step   int
```

Comment:
Basically, the scenario we would like to test was described as:
1) Nothing there
2) conflict (someone else wrote)
3) new value (the data they wrote)

This is translated to: Get() -> Create() -> Get() calls. "step" is simulating what call we are now expecting - so if "step=0" we expect "Get" call, if "step=1" we expect Create, etc.